python/py-tensorflow-tensorboard/Portfile

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-tensorflow-tensorboard
+version             0.4.0rc3
+platforms           darwin
+license             Apache-2
+maintainers         @emcrisostomo openmaintainer
+
+description         TensorFlow helps the tensors flow
+long_description    TensorBoard is a suite of web applications for inspecting \
+                    and understanding your TensorFlow runs and graphs.
+
+homepage            https://github.com/tensorflow/tensorboard
+master_sites        https://pypi.python.org/packages/d5/fb/80df4eb3234c41beeeb3122b1966effbf08608fb80d2fa74bb72f8da9cb3
+
+if {${python.version} < 30} {
+    distname            tensorflow_tensorboard-${version}-py2-none-any
+} else {
+    distname            tensorflow_tensorboard-${version}-py3-none-any
+}
+
+extract.suffix      .whl
+extract.cmd         cp
+extract.pre_args
+extract.post_args   ${worksrcpath}
+extract.mkdir       yes
+
+checksums           rmd160  52c4d0d3bf8856e2ebc87ae9785319e4416ccdce \
+                    sha256  2b8741aed01f35c95b781d25ebf1a84a45c2980746a616a36ade7ed8078d88ee
+
+python.versions     27 34 35 36
+
+if {${name} ne ${subport}} {
+
+    depends_build-append \
+        port:py${python.version}-pip
+
+    depends_lib-append \
+        port:py${python.version}-wheel \
+        port:py${python.version}-bleach \
+        port:py${python.version}-markdown \
+        port:py${python.version}-html5lib \
+        port:py${python.version}-werkzeug \
+        port:py${python.version}-protobuf3 \
+        port:py${python.version}-six \
+        port:py${python.version}-numpy
+
+    if {${python.version} eq 27} {
+        depends_lib-append \
+            port:py${python.version}-futures \
+    }
+
+    build {
+    }
+
+    destroot.cmd        pip-${python.branch}
+    destroot.pre_args
+    destroot.args       \
+        install --no-cache-dir --no-dependencies \
+        --target ${destroot}${python.pkgd} \
```

Comment:
Is this correct? I would have thought you would install into a destroot with `--root=${destroot}`, which is what the python portgroup normally puts into ${destroot.destdir} (which is passed in destroot.post_args except you're clearing that below.)

Reply:
`--root` is the first thing I tried and I hit strange permission errors. I retried after your review and it seems to work. At this point I may have had some garbage, I can't replicate the error now.

Reply:
Ok, it wasn't apparent just changing one port. If I install the `py-tensorflow`port when using `--root=${destroot}` I get this behaviour:

```
% sudo port install py-tensorflow
--->  Computing dependencies for py-tensorflow
The following dependencies will be installed:
 py27-tensorflow
 py27-tensorflow-tensorboard
Continue? [Y/n]: y
--->  Fetching distfiles for py27-tensorflow-tensorboard
--->  Verifying checksums for py27-tensorflow-tensorboard
--->  Extracting py27-tensorflow-tensorboard
--->  Configuring py27-tensorflow-tensorboard
--->  Building py27-tensorflow-tensorboard
--->  Staging py27-tensorflow-tensorboard into destroot
--->  Installing py27-tensorflow-tensorboard @0.4.0rc3_0
--->  Activating py27-tensorflow-tensorboard @0.4.0rc3_0
--->  Cleaning py27-tensorflow-tensorboard
--->  Fetching distfiles for py27-tensorflow
--->  Verifying checksums for py27-tensorflow
--->  Extracting py27-tensorflow
--->  Configuring py27-tensorflow
--->  Building py27-tensorflow
--->  Staging py27-tensorflow into destroot
--->  Installing py27-tensorflow @1.4.1_0
--->  Activating py27-tensorflow @1.4.1_0
Error: Failed to activate py27-tensorflow: Image error: /opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/tensorboard is being used by the active py27-tensorflow-tensorboard port.  Please deactivate this port first, or use 'port -f activate py27-tensorflow' to force the activation.
Error: See /opt/local/var/macports/logs/_Users_enrico_ports_python_py-tensorflow/py27-tensorflow/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets to report a bug.
Error: Processing of port py-tensorflow failed

% sudo port -f activate py27-tensorflow
--->  Computing dependencies for py27-tensorflow
--->  Activating py27-tensorflow @1.4.1_0
Warning: File /opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/tensorboard already exists.  Moving to: /opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/tensorboard.mp_1513806190.
Warning: File /opt/local/bin/tensorboard-2.7 already exists.  Moving to: /opt/local/bin/tensorboard-2.7.mp_1513806190.
--->  Cleaning py27-tensorflow
```

If I use `--target`, then I get this behaviour:

```
% sudo port install py-tensorflow
--->  Computing dependencies for py-tensorflow
The following dependencies will be installed:
 py27-tensorflow
 py27-tensorflow-tensorboard
Continue? [Y/n]: y
--->  Fetching distfiles for py27-tensorflow-tensorboard
--->  Verifying checksums for py27-tensorflow-tensorboard
--->  Extracting py27-tensorflow-tensorboard
--->  Configuring py27-tensorflow-tensorboard
--->  Building py27-tensorflow-tensorboard
--->  Staging py27-tensorflow-tensorboard into destroot
--->  Installing py27-tensorflow-tensorboard @0.4.0rc3_0
--->  Activating py27-tensorflow-tensorboard @0.4.0rc3_0
--->  Cleaning py27-tensorflow-tensorboard
--->  Fetching distfiles for py27-tensorflow
--->  Verifying checksums for py27-tensorflow
--->  Extracting py27-tensorflow
--->  Configuring py27-tensorflow
--->  Building py27-tensorflow
--->  Staging py27-tensorflow into destroot
--->  Installing py27-tensorflow @1.4.1_0
--->  Activating py27-tensorflow @1.4.1_0
--->  Cleaning py27-tensorflow
--->  Fetching distfiles for py-tensorflow
--->  Verifying checksums for py-tensorflow
--->  Extracting py-tensorflow
--->  Configuring py-tensorflow
--->  Building py-tensorflow
--->  Staging py-tensorflow into destroot
--->  Installing py-tensorflow @1.4.1_0
--->  Activating py-tensorflow @1.4.1_0
--->  Cleaning py-tensorflow
--->  Updating database of binaries
--->  Scanning binaries for linking errors
--->  No broken files found.
```

I have no idea why, but I banged my head against the table quite a few times a couple of days ago.